src/KnowbaseItem.php

```diff
@@ -126,6 +126,10 @@ public function canViewItem(): bool
             return true;
         }
 
+        if ($this->fields["is_token_url"] && (isset($_GET['token']) && $this->fields["token"] == $_GET['token'])) {
```

Comment:
When comparing token, we need to use `hash_equals` to prevent timing attacks (now that I think of it I need to also do it for the forms feature).

```suggestion
        if ($this->fields["is_token_url"] && (isset($_GET['token']) && hash_equals($_GET['token'], $this->fields["token"]))) {
```